core/modules/comment/comment.theme.inc

```diff
@@ -156,7 +156,10 @@ function template_preprocess_comment(&$variables) {
   else {
     $variables['classes'][] = 'comment-title-custom';
   }
-  
+
   // Gather classes specific to the comment title.
   $variables['title_classes'][] = 'comment-title';
+
+  // Add extra theme hook suggestions for comment view_mode.
+  $variables['theme_hook_suggestions'][] = 'comment__node_' . $node->type . '__' . $view_mode;
```

Comment:
See my comment above. We'll need to add:

`$variables['theme_hook_suggestions'][] = 'comment__node_' . $node->type . '__' . $view_mode;`

here, in order to make the `comment--node-TYPE` available after modifying the controller as above.

Reply:
I see. However, an important difference is that D7 does not provide the `comment--node-TYPE-VIEWMODE` suggestion, which avoids the problem I mentioned above.

So, then, perhaps we should not change anything related to `comment` at this point, since there is no bug. The code you have (adding the suggestion for `comment--node-TYPE-VIEWMODE`) actually creates a bug rather than solve it, since (with this PR) the `comment--node-TYPE`  template has preference over `comment--node-TYPE-VIEWMODE`